### PR TITLE
fix(dependency): add trufflehog dependency

### LINKED ISSUE #4091: [Bug]: Pre-commit Execution Failure Due to Missing TruffleHog Dependency
### Steps to Reproduce

Commit a new change to start the pre-commit 
`git commit -am "Any commit message"`
Or launch pre-commit to launch trufflehog
`pre-commit run trufflehog `

### Expected behavior

When you install the Prowler dependencies using this guide: [Prowler Developer Guide](https://docs.prowler.com/projects/prowler-open-source/en/latest/developer-guide/introduction/#contributing-with-your-code-or-fixes-to-prowler), the library TruffleHog is not mentioned as a required installation. However, TruffleHog is necessary for certain functionalities. You can find more information about TruffleHog here: [TruffleHog GitHub Repository](https://github.com/trufflesecurity/trufflehog).

Because trufflehog is not installed, it returns an error.

### Actual Result with Screenshots or Logs

TruffleHog...............................................................Failed
- hook id: trufflehog
- exit code: 127

image

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

Workstation

### OS used

MacOs

### Prowler version

Prowler 4.1.0

### Pip version

pip 24.0

### Context

_No response_

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Fixes #4091 by adding Trufflehog as a dev dependency. 

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by MrCloudSec** [comment]
Do not merge until TruffleHog update the GitPython version to a non vulnerable one.

File: pyproject.toml [hunk 1]
```diff
@@ -94,6 +94,7 @@ pytest-env = "1.1.3"
 pytest-randomly = "3.15.0"
 pytest-xdist = "3.6.1"
 safety = "3.2.0"
+trufflehog = "2.2.1"
```
Comment (jfagoagas) [inline pyproject.toml:97]: This is not valid, Trufflehog is a golang package, the Python version is the old version.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
This cannot be merged as is, we need to explictly set that `trufflehog` needs to be installed using https://github.com/trufflesecurity/trufflehog?tab=readme-ov-file#floppy_disk-installation

File: pyproject.toml (hunk 1 above)
Comment (MrCloudSec) [reply pyproject.toml:97]: Closing the PR then.

**Pull request closed without merging**